### DISCUSSION TITLE
[Bug] Fail to install or update some apps

### DISCUSSION THREAD

**Discussion by yanecc** [ScoopInstaller/Scoop#5892] [Help and Questions]
Now I can't update or install any software. The description of the error is quite vague. I have checked the bucket and github token. I am not sure what to do.
Update:
```
~> scoop update -a
beekeeper-studio: 4.1.13 -> 4.2.8
nu: 0.91.0 -> 0.92.1
pnpm: 8.15.5 -> 8.15.6
ttyd: 1.7.4 -> 1.7.7
Updating 4 outdated apps:
Updating 'beekeeper-studio' (4.1.13 -> 4.2.8)
Downloading new version
MetadataError: A:\Scoop\apps\scoop\current\lib\install.ps1:111
Line |
 111 |          throw $e
     |          ~~~~~~~~
     | The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
```
Install:
```
~> scoop install dagger
Installing 'dagger' (0.11.0) [64bit] from main bucket
The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
URL https://github.com/dagger/dagger/releases/download/v0.11.0/dagger_v0.11.0_windows_amd64.zip is not valid
```

**Comment by niheaven** [comment]
Could reproduce now?

**Comment by yanecc** [comment]
> Could reproduce now?

Yes, I still couldn't install anything.

**Comment by niheaven** [comment]
Which branch do you use? `master` or `develop`?

**Comment by yanecc** [comment]
> Which branch do you use? `master` or `develop`?

`master`

**Comment by yanecc** [comment]
Here is my config file
```
{
    "root_path":  "A:\\Scoop",
    "global_path":  "A:\\Scoop\\apps",
    "last_update":  "2024-03-21T12:22:18.6425898+08:00",
    "use_external_7zip":  true,
    "gh_token":  "github_pat_...",
    "scoop_branch":  "master",
    "scoop_repo":  "https://gitee.com/glsnames/scoop-installer"
}
```

**Comment by niheaven** [comment]
Run `scoop config aria2-enabled true` and try again.

**Comment by yanecc** [comment]
> Run `scoop config aria2-enabled true` and try again.

Like magic it works, thank you so much!

**Comment by yanecc** [comment]
> Run `scoop config aria2-enabled true` and try again.

Unfortunately, once again, I found myself unable to install or update any software. My current configuration file is as follows, and even though the option `aria2-enabled` has been set to `true`, the same error appears again.

```
root_path         : A:\Scoop
global_path       : A:\Scoop\apps
last_update       : 2024-03-23T14:01:50.2268783+08:00
use_external_7zip : True
gh_token          : github_pat_...
aria2-enabled     : True
scoop_branch      : master
scoop_repo        : https://gitee.com/glsnames/scoop-installer
```

**Comment by niheaven** [comment]
Then it may be a network error now... What's the error message?

**Comment by yanecc** [comment]
> Then it may be a network error now... What's the error message?

The error is exactly the same as before.😫

```
Installing 'dagger' (0.11.0) [64bit] from main bucket
The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
URL https://github.com/dagger/dagger/releases/download/v0.11.0/dagger_v0.11.0_windows_amd64.zip is not valid
```

**Comment by chawyehsu** [comment]
Apparently `https://gitee.com/glsnames/scoop-installer` is a modified source of Scoop..., can you try `scoop config scoop_repo https://github.com/ScoopInstaller/Scoop`?

**Comment by yanecc** [comment]
> Apparently `https://gitee.com/glsnames/scoop-installer` is a modified source of Scoop..., can you try `scoop config scoop_repo https://github.com/ScoopInstaller/Scoop`?

Tried, got the same result.

**Comment by yanecc** [comment]
Thank all of you for your advice!
It's cheerful for me to see that the problem was resolved after upgrading to version 0.4.0.

**Comment by tangturtle** [comment]
A few days ago, I encountered the same issue. After troubleshooting, I discovered that it was related to the `gh_token`. Previously, I had set a `gh_token` because the number of `scoop search` requests had reached its limit. Immediately after that, the issue appeared. When I removed the `gh_token` by running `scoop config gh_token ""`, the problem was resolved.
